backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/utils/CassandraAsyncExecutor.java

```diff
@@ -76,4 +80,23 @@ public CassandraAsyncExecutor(Session session) {
         return executeSingleRow(statement)
                 .hasElement();
     }
+
+    private Mono<ResultSet> retryOnError(Mono<ResultSet> query) {
+        return query
+                .retryWhen(companion -> companion
```

Comment:
why not the retry backoff ? :)

Reply:
It's not few lines after:

```
                 .flatMap(index -> Mono.delay(Duration.ofMillis(index * 100)))
```

You have to filter on the `Exception` class, that's why I've used a more general operator.